clusterloader2/docs/test_suite_design.md

```diff
@@ -0,0 +1,46 @@
+# Cluster loader - TestSuite
+
+## Background
+
+Since 08/2018 cluster loader become main tool for kubernetes e2e performance tests.
+It's is a very convenient tool for running single test case, but not for running
+test bundles. Currently it is impossible to run the same test
+with two different parameters sets in a single job run.
+This forces developers to create separate job configs for
+almost each tests.
+
+## TestSuite
+
+TestSuite is a test bundle.
+Overrides is a set of override file paths. Overrides provided on the TestSuite level will be applied
+to all of the test, unless they are overidden on the test level.
+
+```
+type TestSuite struct {
+	Overrides []string
+	Tests     []Test
+}
+
+```
+
+## Test
+
+Test defines a single test scenario. Config defines path to the config file
+that satisfy already existing cluster loader API. Overrides defines what
+override files should be applied to the config file.
+To enable using same test config in one test suite,
+the Identifier field was introduced. This indetifier will be used to match
```

Comment:
So is it only used for reporting or for something else?

Reply:
IMHO user will only see it as something for matching results. We can still implement it without identifier, however user will have to match results with tests based on timestamp. This doesn't look user-friendly.